streaming/src/queue/transport.h

```diff
@@ -0,0 +1,256 @@
+  ObjectID queue_id_;
+  std::shared_ptr<LocalMemoryBuffer> buffer_;
+
+ public:
+  static const uint32_t MagicNum;
+};
+
+class DataMessage : public Message {
+ public:
+  DataMessage(const ActorID &actor_id, const ActorID &peer_actor_id, ObjectID queue_id,
+              uint64_t seq_id, std::shared_ptr<LocalMemoryBuffer> buffer, bool raw)
+      : Message(actor_id, peer_actor_id, queue_id, buffer), seq_id_(seq_id), raw_(raw) {}
+  virtual ~DataMessage() {}
+
+  static std::shared_ptr<DataMessage> FromBytes(uint8_t *bytes);
+  virtual void ConstructFlatBuf(flatbuffers::FlatBufferBuilder &builder);
+
+  uint64_t SeqId() { return seq_id_; }
+  bool IsRaw() { return raw_; }
+  queue::flatbuf::MessageType Type() { return type_; }
+
+ private:
+  uint64_t seq_id_;
+  bool raw_;
+
+  const queue::flatbuf::MessageType type_ =
+      queue::flatbuf::MessageType::StreamingQueueDataMsg;
+};
+
+class NotificationMessage : public Message {
+ public:
+  NotificationMessage(const ActorID &actor_id, const ActorID &peer_actor_id,
+                      const ObjectID &queue_id, uint64_t seq_id)
+      : Message(actor_id, peer_actor_id, queue_id), seq_id_(seq_id) {}
+
+  virtual ~NotificationMessage() {}
+
+  static std::shared_ptr<NotificationMessage> FromBytes(uint8_t *bytes);
+  virtual void ConstructFlatBuf(flatbuffers::FlatBufferBuilder &builder);
+
+  uint64_t SeqId() { return seq_id_; }
+  queue::flatbuf::MessageType Type() { return type_; }
+
+ private:
+  uint64_t seq_id_;
+  const queue::flatbuf::MessageType type_ =
+      queue::flatbuf::MessageType::StreamingQueueNotificationMsg;
+};
+
+class CheckMessage : public Message {
+ public:
+  CheckMessage(const ActorID &actor_id, const ActorID &peer_actor_id,
+               const ObjectID &queue_id)
+      : Message(actor_id, peer_actor_id, queue_id) {}
+  virtual ~CheckMessage() {}
+
+  static std::shared_ptr<CheckMessage> FromBytes(uint8_t *bytes);
+  virtual void ConstructFlatBuf(flatbuffers::FlatBufferBuilder &builder);
+
+  queue::flatbuf::MessageType Type() { return type_; }
+
+ private:
+  const queue::flatbuf::MessageType type_ =
+      queue::flatbuf::MessageType::StreamingQueueCheckMsg;
+};
+
+class CheckRspMessage : public Message {
+ public:
+  CheckRspMessage(const ActorID &actor_id, const ActorID &peer_actor_id,
+                  const ObjectID &queue_id, queue::flatbuf::StreamingQueueError err_code)
+      : Message(actor_id, peer_actor_id, queue_id), err_code_(err_code) {}
+  virtual ~CheckRspMessage() {}
+
+  static std::shared_ptr<CheckRspMessage> FromBytes(uint8_t *bytes);
+  virtual void ConstructFlatBuf(flatbuffers::FlatBufferBuilder &builder);
+  queue::flatbuf::MessageType Type() { return type_; }
+  queue::flatbuf::StreamingQueueError Error() { return err_code_; }
+
+ private:
+  queue::flatbuf::StreamingQueueError err_code_;
+  const queue::flatbuf::MessageType type_ =
+      queue::flatbuf::MessageType::StreamingQueueCheckRspMsg;
+};
+
+class TestInitMsg : public Message {
```

Comment:
```suggestion
class TestInitMessage : public Message {
```
Using abbreviations is okay, but we should be consistent about naming these message types.

Reply:
> Using abbreviations is okay, but we should be consistent about naming these message types.

fixed.